wagtail/contrib/modeladmin/views.py

```diff
@@ -798,20 +807,23 @@ def post(self, request, *args, **kwargs):
             return redirect(self.index_url)
         except models.ProtectedError:
             linked_objects = []
-            fields = self.model._meta.fields_map.values()
-            fields = (obj for obj in fields if not isinstance(
-                obj.field, ManyToManyField))
-            for rel in fields:
+            for rel in self._get_related_fields():
                 if rel.on_delete == models.PROTECT:
                     if isinstance(rel, OneToOneRel):
                         try:
-                            obj = getattr(self.instance, rel.get_accessor_name())
+                            if rel.get_accessor_name() != '+':
```

Comment:
Arguably, the performance of both `if` and `else` OneToOneField implementations are similar, they're both a "get" operation, so we could potentially drop the `if` and use the `else` implementation all the time...

In contrast, for the implementation for ManyToOneFields, I assume a releated_name lookup is more efficient than doing a filter of the related model? In other words, I imagine leaving the `if` and `else` implementations makes sense for ManyToOneField.